rest/blip_client_test.go

```diff
@@ -484,23 +531,85 @@ func createBlipTesterClientOpts(tb testing.TB, rt *RestTester, opts *BlipTesterC
 	btc := BlipTesterClient{
 		BlipTesterClientOpts: *opts,
 		rt:                   rt,
-		docs:                 make(map[string]map[string]*BodyMessagePair),
-		attachments:          make(map[string][]byte),
-		lastReplicatedRev:    make(map[string]string),
 	}
 
 	id, err := uuid.NewRandom()
 	if err != nil {
 		return nil, err
 	}
 
-	if btc.pushReplication, err = newBlipTesterReplication(tb, "push"+id.String(), &btc); err != nil {
+	if btc.pushReplication, err = newBlipTesterReplication(btc.rt.tb, "push"+id.String(), &btc); err != nil {
 		return nil, err
 	}
-	if btc.pullReplication, err = newBlipTesterReplication(tb, "pull"+id.String(), &btc); err != nil {
+	if btc.pullReplication, err = newBlipTesterReplication(btc.rt.tb, "pull"+id.String(), &btc); err != nil {
 		return nil, err
 	}
 
+	btc.CollectionClients = make(map[string]*BlipTesterCollectionClient)
+
+	if len(opts.Collections) != 0 {
+		getCollectionsRequest := blip.NewRequest()
+		getCollectionsRequest.SetProfile(db.MessageGetCollections)
+
+		type CollectionsRequest struct {
+			Collections   []string `json:"collections"`
+			CheckpointIDs []string `json:"checkpoint_ids"`
+		}
+
+		checkpointIDs := make([]string, len(btc.Collections))
+		for i := range checkpointIDs {
+			checkpointIDs[i] = btc.pullReplication.id
+		}
+
+		requestBody := CollectionsRequest{
+			Collections:   btc.Collections,
+			CheckpointIDs: checkpointIDs,
+		}
+		body, err := base.JSONMarshal(requestBody)
+		if err != nil {
+			return nil, err
+		}
+
+		getCollectionsRequest.SetBody(body)
+		if err := btc.pullReplication.sendMsg(getCollectionsRequest); err != nil {
+			return nil, err
+		}
+
+		type CollectionsResponseEntry struct {
+			LastSequence *int    `json:"last_sequence"`
+			Rev          *string `json:"rev"`
+		}
```

Comment:
If this is the response, I wonder if we should change blip_api_collections_test.go to use this instead of `db.Body` which is `map[string]interface{}`